platform/linux-generic/pktio/uapi_net_mdev.h

```diff
@@ -0,0 +1,33 @@
+/* Copyright (c) 2018, Linaro Limited
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+#ifndef _UAPI__LINUX_NET_MDEV_H
+#define _UAPI__LINUX_NET_MDEV_H
+
+#include <linux/types.h>
+#include <linux/ioctl.h>
+
+enum net_mdev_types {
+	VFIO_NET_MDEV_SHADOW,
+	VFIO_NET_DESCRIPTORS,
+	VFIO_NET_MMIO,
+};
+
+enum net_mdev_subtypes {
+	VFIO_NET_MDEV_STATS,
+	VFIO_NET_MDEV_RX,
+	VFIO_NET_MDEV_TX,
+	VFIO_NET_MDEV_BARS,
+};
```

Comment:
Subtypes of which type? Or are all subtypes applicable to all types?

Reply:
At the moment MMIO is using subtype BARS and DESCRIPTORS are using Rx/Tx
Shadow is unused

Reply:
I then suggest to rename subtypes, so that they will reference the main type.

Reply:
like `VFIO_NET_MDEV_MMIO_BARS`, `V_N_M_DESCRIPTORS_RX`, etc.

Reply:
Agree with @lumag that having names the reflect the hierarchy will be helpful in avoiding errors.

Reply:
Ok, will be addressed in the patch revision.